src/main/java/de/dennisguse/opentracks/TrackEditActivity.java

```diff
@@ -65,13 +62,17 @@ protected void onCreate(Bundle bundle) {
         }
 
         contentProviderUtils = new ContentProviderUtils(this);
-        track = contentProviderUtils.getTrack(trackId);
-        if (track == null) {
-            Log.e(TAG, "No track for " + trackId.getId());
-            finish();
-            return;
-        }
+        ContentProviderUtils.RunOutUIThread.build(contentProviderUtils).getTrack(trackId).observe(this, track -> {
```

Comment:
We need to do something about the syntax: 
```
ContentProviderUtils.RunOutUIThread.build(contentProviderUtils)
```

a contentProviderUtils that gets a contentProviderUtils?
Confusing ;)